### ISSUE TITLE
SAINT pre-processing

### ISSUE THREAD

**Issue by Mira0507** [bornea/APOSTL#8]
Hi APOSTL team, 

I got some questions while learning the APOSTL using the test server. 

1. My usage is currently 26% and I'd like to re-upload my input datasets (mzid files). After permanently depleting my files and history, it's still 26%. Is it possible to re-start the pipeline from 0%? 

2. My SAINT pre-processing step raise errors in generating interaction, prey, and bait files even though the input mzid files were successfully uploaded. The "bait create" tab might be the issue causing the errors. If I have 6 input files shown below:
- FLAG_BR1.mzid (negative control, rep1)
- FLAG_BR2.mzid (negative control, rep2)
- BirA_BR1.mzid (negative control, rep1)
- BirA_BR2.mzid (negative control, rep2)
- SAR1A_BR1.mzid (bait, rep1)
- SAR1A_BR2.mzid (bait, rep2)
Would you have an idea about how to correctly fill the blanks "Bait", "Group Assignment", and "is this control" for my datasets? Also, would you have any other suspicions about why it failed? 

3. Is it possible that some mzIdentML files are incompatible with the APOSTL depending on how it's generated (e.g. software version)? 

Thanks,

Mira

**Comment by pstew** [comment]
@Mira0507 Sorry for the delay in responding. 

1) You can access your other histories from the top of the history panel on the right in Galaxy. Once you see all of your histories, you will need to select the option to "permanently delete" a history because something "deleted" in Galaxy is not really removed.

2) (& 3) Unfortunately I have no experience using mzid files with APOSTL. I will ask @brentkuenzi or @bornea to comment here. I can say that we/Galaxy does not do file validation by default, so upload of the files may not mean they are the right version or formatted correctly.

**Comment by brentkuenzi** [comment]
I think the most likely thing is the bait create. Some people have had
issues in the past. I would try preparing your own bait file and uploading
it

Are you uploading mzid files or mzidentml files? Mzid files are different
and won’t work

**Comment by Mira0507** [comment]
Hi @pstew and @brentkuenzi,

Thank you very much for the comments. I figured out that the history of my galaxy instance works correctly with `APOSTL` locally installed. 

I thought mzidentml is identical to mzid. Before learning they are different, I uploaded mzid files to run `SAINT pre-processing`. Turned out that the error was raised by the argument`infile` which was unassigned in the function `bait_create` from `mzID_precoess2.py`. My error has been resolved by removing the `infile`. Eventually it ran error-free with mzid files (but ended up generating many missing values in the prey file). I was able to run `SAINTexpress` with some additional data cleaning of the input files. 

Thank you again!

Best,

Mira
